source/extensions/omni.isaac.lab/omni/isaac/lab/ui/widgets/manager_live_visualizer.py

```diff
@@ -0,0 +1,297 @@
+
+    def _set_vis_frame_impl(self, frame: omni.ui.Frame):
+        """Updates the assigned frame that can be used for visualizations.
+
+        Args:
+            frame: The debug visualization frame.
+        """
+        self._vis_frame = frame
+
+    def _debug_vis_callback(self, event):
+        """Callback for the debug visualization event."""
+
+        if not SimulationContext.instance().is_playing():
+            # Visualizers have not been created yet.
+            return
+
+        # get updated data and update visualization
+        for (_, term), vis in zip(
+            self._manager.get_active_iterable_terms(env_idx=self._env_idx), self._term_visualizers
+        ):
+            if isinstance(vis, LiveLinePlot):
+                vis.add_datapoint(term)
+            elif isinstance(vis, ImagePlot):
+                vis.update_image(numpy.array(term))
+
+    def _set_debug_vis_impl(self, debug_vis: bool):
+        """Set the debug visualization implementation.
+
+        Args:
+            debug_vis: Whether to enable or disable debug visualization.
+        """
+
+        if not hasattr(self, "_vis_frame"):
+            raise RuntimeError("No frame set for debug visualization.")
+
+        # Clear internal visualizers
+        self._term_visualizers = []
+        self._vis_frame.clear()
+
+        if debug_vis:
+            # if enabled create a subscriber for the post update event if it doesn't exist
+            if not hasattr(self, "_debug_vis_handle") or self._debug_vis_handle is None:
+                app_interface = omni.kit.app.get_app_interface()
+                self._debug_vis_handle = app_interface.get_post_update_event_stream().create_subscription_to_pop(
+                    lambda event, obj=weakref.proxy(self): obj._debug_vis_callback(event)
+                )
+        else:
+            # if disabled remove the subscriber if it exists
+            if self._debug_vis_handle is not None:
+                self._debug_vis_handle.unsubscribe()
+                self._debug_vis_handle = None
+
+            self._vis_frame.visible = False
+            return
+
+        self._vis_frame.visible = True
+
+        with self._vis_frame:
+            with omni.ui.VStack():
+                # Add a plot in a collapsible frame for each term available
+                for name, term in self._manager.get_active_iterable_terms(env_idx=self._env_idx):
+                    if name in self.term_names or len(self.term_names) == 0:
+                        frame = omni.ui.CollapsableFrame(
+                            name,
+                            collapsed=False,
+                            style={"border_color": 0xFF8A8777, "padding": 4},
+                        )
+                        with frame:
+                            # create line plot for single or multivariable signals
+                            len_term_shape = len(numpy.array(term).shape)
+                            if len_term_shape <= 2:
```

Comment:
We only want single floats and lists to be for the line plot, so that should be < 2 and then the image for everything that is 2D and 3D. So there is still something wrong with that check